runelite-client/src/main/java/net/runelite/client/plugins/microbot/holidayevent/CollectSnow.java

```diff
@@ -0,0 +1,28 @@
+package net.runelite.client.plugins.microbot.holidayevent;
+
+import net.runelite.api.events.GameTick;
+import net.runelite.client.plugins.microbot.Microbot;
+import net.runelite.client.plugins.microbot.util.gameobject.Rs2GameObject;
+import net.runelite.client.plugins.microbot.util.math.Rs2Random;
+import net.runelite.client.plugins.microbot.util.player.Rs2Player;
+
+import java.awt.event.KeyEvent;
+
+public class CollectSnow {
+    private static long randomDelay;
+
+    public static boolean nearTheSnow() {
+        return Rs2GameObject.findObjectByIdAndDistance(5067, 1) != null;
+    }
+
+    public static void onGameTick(GameTick event) {
```

Comment:
any reason to leverage onGameTick here? I think it would do justice scanning GameObjects & checking player location by making this class extend Script

Reply:
Just saying that these actions can take a long period of time & may not be best to do on client thread

Reply:
I just copied QoL NeverLogOut logic and replaced KeyPress SPACE_BAR with clicking on the snow pile. I figured if webWalker was once on client thread and pressing spacebar is fine on client thread that clicking a game object that you must be 1 tile from for it to trigger would be OK.

Reply:
Just tested everything and works fine on client thread doesn't even stutter or nuttin :D